control-base/src/main/java/fi/nls/oskari/control/layer/SaveLayerHandler.java

```diff
@@ -494,77 +488,33 @@ private void validateInsertLayer(final ActionParameters params, OskariLayer ml)
         }
 
     }
-    private boolean handleWMSSpecific(final ActionParameters params, OskariLayer ml) throws ActionException {
 
-        HttpServletRequest request = params.getRequest();
-        final String xslt = request.getParameter("xslt");
+    private boolean handleWMSSpecific(final ActionParameters params, OskariLayer ml) {
+        final String xslt = params.getHttpParam("xslt");
```

Comment:
The difference here is whether we want to run the value through jsoup or not. Have you tried that xslt is not modified by jsoup?